### DISCUSSION TITLE
Make some types from persist midleware exportable

### DISCUSSION THREAD

**Discussion by chiralium** [pmndrs/zustand#2221] [Ideas] [answered]
![image](https://github.com/pmndrs/zustand/assets/61596633/44680725-7293-4b6f-b1e1-a389d14b370e)

For some reasons it would be greate to export those types. Today I'm was trying to create wrapper outside the original persist function and I've get troubles with the types. 

By the way, I cant understand why this types (on images) not available to export.

**Comment by dai-shi** [comment, marked as answer]
It's intentional. WithFoo type is for module argumentation, and not something that we specify from outside.

If you want to create your own middleware/wrapper that has precise typing, you need to follow the TypeScript guide, and the guide could be improved.
On the other hand, if we want to bypass some typings, it's probably possible with assertion -> #2222.
